ArduinoHallEffect/ArduinoHallEffect.ino

```diff
@@ -0,0 +1,70 @@
+// An Arduino program for controlling valve timing on a 6.5ph Predator engine from Harbor Freight.
+//
+// Wesley Kagan, 2020
+// Ric Allinson, 2020
+
+// Control variables.
+const int DEG_PER_MAGNET = 6;       // Number of degrees for each magnet.
+const int FREEVALVE_OFFSET_TOP = 0; // The amount of off set from TDC (needs to be multiples of DEG_PER_MAGNET).
+const int FREEVALVE_OFFSET_BOTTOM = 180 + FREEVALVE_OFFSET_TOP; 
```

Comment:
Separate variables are necessary for intake/exhaust open/close angles (4 distinct values) as 4 stroke engine cam phases usually are not symmetrical mostly due to flow inertia (you can look them up in google - "camshaft timing chart").
For engine with free phases separate values are even more important - they will be changed by some algorithm depending on RPM and other values from sensors and lookup tables.

Reply:
Also solenoid inertia should be coded or looked up in some table, I think. Similarly to how ECUs have injector opening delay lookup tables depending on system voltage and RPM ..
It will be fun to code as it depends on opening/closing rate and is "real time" not RPM/angle value :)

Not in scope of this PR though :)

Reply:
Super feedback. Thanks. I've started on another branch to work in the intake/exhaust open/close. Just saw @Tamaren1 latest video. Exciting to see how the engine runs after mapping.